airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+# under the License.
+#
+from copy import deepcopy
+from datetime import date, time
+
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_transfer_hook import GCPTransferServiceHook, \
+    GcpTransferJobsStatus
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+try:
+    from airflow.contrib.hooks.aws_hook import AwsHook
+except ImportError:  # pragma: no cover
+    AwsHook = None
+
+
+class TransferJobPreprocessor:
+
+    def __init__(self, body, aws_conn_id='aws_default'):
+        self.body = body
+        self.aws_conn_id = aws_conn_id
+
+    def _inject_aws_credentials(self):
+        if 'transferSpec' not in self.body or \
+           'awsS3DataSource' not in self.body['transferSpec']:
+            return
+
+        aws_hook = AwsHook(self.aws_conn_id)
+        aws_credentials = aws_hook.get_credentials()
+        aws_access_key_id = aws_credentials.access_key
+        aws_secret_access_key = aws_credentials.secret_key
+        self.body['transferSpec']['awsS3DataSource']["awsAccessKey"] = {
+            "accessKeyId": aws_access_key_id,
+            "secretAccessKey": aws_secret_access_key
+        }
+
+    def _reformat_date(self, field_key):
+        if field_key not in self.body['schedule']:
+            return
+        if isinstance(self.body['schedule'][field_key], date):
+            self.body['schedule'][field_key] = self.\
+                _convert_date(self.body['schedule'][field_key])
+
+    def _reformat_time(self, field_key):
+        if field_key not in self.body['schedule']:
+            return
+        if isinstance(self.body['schedule'][field_key], time):
+            self.body['schedule'][field_key] = self.\
+                _convert_time(self.body['schedule'][field_key])
+
+    def _reformat_schedule(self):
+        if 'schedule' not in self.body:
+            return
+        self._reformat_date('scheduleStartDate')
+        self._reformat_date('scheduleEndDate')
+        self._reformat_time('startTimeOfDay')
+
+    def process_body(self):
+        self._inject_aws_credentials()
+        self._reformat_schedule()
+        return self.body
+
+    @staticmethod
+    def _convert_date(field_date):
+        return {
+            'day': field_date.day,
+            'month': field_date.month,
+            'year': field_date.year
+        }
+
+    @staticmethod
+    def _convert_time(time):
+        return {
+            "hours": time.hour,
+            "minutes": time.minute,
+            "seconds": time.second,
+        }
+
+
+class TransferJobValidator:
+
+    def __init__(self, body):
+        self.body = body
+
+    def _verify_data_source(self):
+        if 'transferSpec' not in self.body:
+            return
+
+        is_gcs = 'gcsDataSource' in self.body['transferSpec']
+        is_aws_s3 = 'awsS3DataSource' in self.body['transferSpec']
+        is_http = 'httpDataSource' in self.body['transferSpec']
+
+        if not is_gcs ^ is_aws_s3 ^ is_http:
```

Comment:
What happens if we specify all three sources ?.Here are some suggestions as python has no builtin for this it seems:  https://stackoverflow.com/questions/1032411/check-if-only-one-variable-in-a-list-of-variables-is-set

Reply:
@potiuk Good point. I will fix it.

Reply:
Fixed.